extensions/lem-dashboard/dashboard-items.lisp

```diff
@@ -0,0 +1,147 @@
+  ((display-text :initarg :display-text :accessor display-text))
+  (:documentation "Creates a link/button with DISPLAY-TEXT that executes the keybind-command.")
+  (:default-initargs
+   :item-attribute 'document-text-attribute))
+
+(defmethod initialize-instance :after ((item dashboard-command) &key)
+  (with-slots (keybind-command action) item
+    (setf action (lambda () (funcall keybind-command)))))
+
+(defmethod draw-dashboard-item ((item dashboard-command) point)
+  (lem/button:insert-button point 
+                            (create-centered-string (display-text item))
+                            (lambda () (funcall (keybind-command item) item))
+                            :attribute (item-attribute item)))
+
+;; Recent projects
+(defclass dashboard-recent-projects (dashboard-item)
+  ((project-count :initarg :project-count :accessor project-count :initform *dashboard-project-count*))
+  (:documentation "Displays a list of recent projects, limited to the last PROJECT-COUNT.")
+  (:default-initargs
+   :item-attribute 'document-text-attribute
+   :action (lambda ()
+             (let ((project (str:trim (line-string (current-point)))))
+               (when project
+                 (lem-core/commands/project:project-find-file project))))))
+
+(define-command move-to-recent-projects () ()
+  (let ((point (buffer-point (current-buffer))))
+    (buffer-start point)
+    (search-forward-regexp point "Recent Projects")
+    (line-offset point 2)
+    (move-to-beginning-of-line)))
+
+(defmethod draw-dashboard-item ((item dashboard-recent-projects) point)
+  (let* ((title (format nil "~A Recent Projects (r)" (icon-string "package")))
+         (title-line (create-centered-string title)))
+    (insert-string point title-line :attribute 'document-header1-attribute)
+    (insert-character point #\Newline)
+    (insert-character point #\Newline)
+    (let* ((projects (reverse (lem-core/commands/project:saved-projects)))
+           (longest-project (reduce #'(lambda (a b) (if (> (length a) (length b)) a b)) projects))
+           (max-length (length longest-project))
+           (left-padding (floor (- (window-width (current-window)) max-length) 2)))
+      (loop for project in (subseq projects 0 (min (project-count item) (length projects)))
+            do (insert-string point (format nil "~v@{~A~:*~}" left-padding " "))
+               (insert-string point (format nil "~A~%" project))))))
+
+;; Recent files
+(defclass dashboard-recent-files (dashboard-item)
+  ((file-count :initarg :file-count :accessor file-count :initform *dashboard-file-count*))
+  (:documentation "Displays a list of recent files, limited to the last FILE-COUNT.")
+  (:default-initargs
+   :item-attribute 'document-text-attribute
+   :action (lambda ()
+             (let ((file (str:trim (line-string (current-point)))))
+               (when file
+                 (find-file file))))))
+
+(define-command move-to-recent-files () ()
```

Comment:
Since define-command has a global namespace, it would be better to have a dedicated prefix.